File: movies-api/src/main/java/com/ivanfranchin/moviesapi/exception/MovieNotFoundException.java
package com.ivanfranchin.moviesapi.exception;

import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ResponseStatus;

@ResponseStatus(HttpStatus.NOT_FOUND)
public class MovieNotFoundException extends RuntimeException {

    public MovieNotFoundException(String imdbId) {
        super("Movie with imdbId '%s' not found".formatted(imdbId));
    }
}

File: movies-api/src/main/kotlin/com/ivanfranchin/moviesapi/movie/exception/MovieNotFoundException.kt
package com.ivanfranchin.moviesapi.movie.exception

import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ResponseStatus

@ResponseStatus(HttpStatus.NOT_FOUND)
class MovieNotFoundException(imdbId: String?) :
    RuntimeException("Movie with imdbId $imdbId not found")